packages/nimble-components/src/table/tests/table-column-sizing.spec.ts

```diff
@@ -46,6 +46,28 @@ async function setup(): Promise<Fixture<Table<SimpleTableRecord>>> {
     );
 }
 
+async function growTableToGivenRowWidth(
```

Comment:
Some clean code suggestions for this function:
- I don't love the name since it doesn't enforce that the width always grows (vs shrinks) and since "row width" to me implies setting the width of all rows. But I don't have great suggestions for a better name (possibly indicating the function's purpose isn't very crisp). Best I can think of is something like `setTableWidthWithPadding`. (related: should this reserve space for other chrome like the multiselection boxes?)
- consider moving the `querySelector` and button size logic to a page object function like `getCollapseButtonWidth()`

Reply:
I'll admit that I have mixed feelings about moving anything using a `querySelector` to the page object. Ultimately, I think we want the page objects to serve as the means for clients to write tests where they can interact with the table in ways which the API doesn't afford (like clicking buttons, etc...). I don't expect that a client should need to write a test that hinges on the size of the collapse-all button (and if they do, that might be a sign of a bad smell). That said, I don't suppose it hurts anything either with it being there, except for API bloat.

Reply:
I _did_ update the name to `sizeTableToGivenRowWidth`. Regarding your question about reserving space for other chrome, I believe the answer is "no", as those elements do not have reserved space, and this test file hasn't, yet, exercised the features that would cause those elements to appear. It's quite possible that that would be good to do, but I prefer to hold off on making changes to support that until we write tests that exercise that state.

Reply:
I guess the purpose of the table page objects hasn't really crystallized for me yet. So far we've been adding whatever methods our tests need without much intentional design of the client-facing API (the table page object has nearly 40 public methods on it already). I'm not opposed to a more thoughtful API but we should probably tackle that holistically with an audit of the existing API and some documentation about our philosophy.

But in the absence of any policy I don't feel too strongly about my suggestion so I'm fine if you don't think it should be moved.

Reply:
My current thoughts on page objects are they are intended to make it so changes in the UI are not brittle for tests kinda following the [Martin Fowler PageObject](https://martinfowler.com/bliki/PageObject.html) mindset. So this method does do that and I think makes sense in a PageObject.

I could see us maybe having separate page objects, one that is part of a public facing testing api and one that is internal only, but conceptually I think it makes sense in a page object.